code/modules/power/apc/apc_appearance.dm

```diff
@@ -116,3 +106,19 @@
 // Used in process so it doesn't update the icon too much
 /obj/machinery/power/apc/proc/queue_icon_update()
 	icon_update_needed = TRUE
+
+// Shows a dark-blue interface for a moment. Shouldn't appear on cameras.
+/obj/machinery/power/apc/proc/flicker_hacked_icon()
+	var/image/hacker_image = image(icon = 'icons/obj/machines/wallmounts.dmi', loc = src, icon_state = "apcemag", layer = FLOAT_LAYER)
+	var/list/mobs_to_show = list()
+	for(var/mob/viewer in view(src))
+		if(viewer.client)
+			mobs_to_show += viewer.client
+		if(istype(viewer, /mob/camera/ai_eye))
+			var/mob/camera/ai_eye/viewer_eye = viewer
+			if(viewer_eye.ai == malfai && viewer_eye.ai.client)
+				mobs_to_show |= viewer_eye.ai.client
```

Comment:
If the sole intent of this code is to ensure ONLY the ai that has hacked it can see the animatoin, it's not necessary / the next block of code does this
```suggestion
```

Reply:
This code, and the following, have been removed in favor of client_image_holder objects for the AI.